### ISSUE TITLE
Can I make operations on visual lines work like on non-visual lines?

### ISSUE THREAD

**Issue by thomasheartman** [emacs-evil/evil#1442]
# Issue type

- Question

# Question

Is it possible to make operations on visual lines, such as `ygj` and `dgj` work in a similar manner to what they would on logical lines? I know this is different from how Vim works, so it's not a bug, just a preference.

I'm used to `dj` and `dk` (or `yj`, `cj`, etc) to operate on multiple logical lines. Deletion is most common, so that's what I'll be referring to here, but the others also happen. When using `evil-next-line` and `evil-previous-line`, Evil deletes the whole current (logical) line and the whole next/previous line. When using `evil-next-visual-line` and `evil-previous-visual-line`, it deletes from point to the same column on the next/previous line, whether that is the same logical line or not.

I was wondering whether it's possible to make Evil act on the whole visual line in this case to mirror the behavior on logical lines. 

In other words, I want this (where `|` is the cursor):
```
some| line
another line
final line
```
followed by `dgj` to be
```
final line
```

Instead, it is
```
someher line
final line
```

In the event that the lines are only visual, I'd ideally see the same kind of effect.
If the below were a single line of text that is visually wrapped (and `|` the cursor again):
```
Lorem ipsum| dolor sit amet,
consectetur adipiscing elit.
Integer consectetur urna at.
```

I'd like `dgj` to result in this:
```
Integer consectetur urna at.
```

Can this be done in some way?

Thanks!

# Environment

Emacs version: GNU Emacs 28.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.24, cairo version 1.16.0)
Operating System: NixOS 20.09
Evil version: Evil version 1.14.0
Evil installation type: MELPA
Graphical/Terminal: Graphical
Tested in a `make emacs` session (see [CONTRIBUTING.md](https://github.com/emacs-evil/evil/blob/master/CONTRIBUTING.md#bug-reports)):  No

**Comment by tomdl89** [comment]
I don't think this would be trivial to do, but of course it is possible. `evil-next-line` and `evil-previous-line` have a type of `exclusive` so you'd want to make versions of these with a new type which expanded to visual line(s) instead.

**Comment by thomasheartman** [comment]
Hey 🙋🏾  Sorry for the slow response. Finally got around to having a little look at this. 

From what I can tell `evil-next-line` and `evil-previous-line` have `:type line` , while their visual counterparts have `:type exclusive`, but I'm assuming that's what you meant 😅  Anyway, if I understand you correctly, then what you're suggesting is I create a new type using the `evil-define-type` macro and create replacements for next/previous visual line functions that are of this new type. Is that correct?

I had a brief look at the existing type definitions now and it looks like a little bit of work, but definitely not impossible. I'll see what I can do about it. Thanks for the input! :pray:

**Comment by tomdl89** [comment]
>...but I'm assuming that's what you meant...

Yes, sorry that's exactly what I meant. And the approach of creating a new type would indeed be where I'd first look. But having just taken another look, there is a `screen-line` type already - maybe that's (similar to) what you want? Keep us posted on where you get to with this. It may be worth a PR if it can be integrated without adding too much complexity.

**Comment by thomasheartman** [comment]
Yeah, I noticed the `screen-line` type too. I'll have to look a bit more into it because I'm not entirely sure how I want it to work yet 😅  I use a combination of visual and logical lines which complicates things. For instance, I've remapped next and prev line commands to their visual counterparts, but I still use Emacs' `C-a` and `C-e` (and Evil's `I` and `A`) to operate on the logical line, even when wrapped, so I need to experiment a bit with it a bit. 

I _think_ what I want is for them to act like visual line movements except for in operator pending state, where I want them to operate on the visual line 🤔 

Thanks again for your thoughts! I'll try and keep you up to date with what I find.
